dsgrid/registry/project_registry_manager.py

```diff
@@ -642,8 +616,158 @@ def _submit_dataset_and_register_mappings(
                     raise DSGValueNotRegistered(f"mapping_id={ref.mapping_id}")
                 references.append(ref)
 
+        references += self._auto_register_reverse_supplemental_mappings(
+            project_config, dataset_config, references, submitter, log_message, context
+        )
         self._submit_dataset(project_config, dataset_config, submitter, log_message, references)
 
+    def _register_mappings_from_file(
+        self,
+        project_config,
+        dataset_config,
+        dimension_mapping_file,
+        submitter,
+        log_message,
+        context,
+    ):
+        references = []
+        src_dir = dimension_mapping_file.parent
+        mappings = DatasetBaseToProjectMappingTableListModel(
+            **load_data(dimension_mapping_file)
+        ).mappings
+        dataset_mapping = {x.dimension_type: x for x in dataset_config.model.dimension_references}
+        project_mapping = {
+            x.dimension_type: x for x in project_config.model.dimensions.base_dimension_references
+        }
+        mapping_tables = []
+        for mapping in mappings:
+            mapping_table = run_in_other_dir(
+                src_dir,
+                MappingTableModel.from_pre_registered_model,
+                mapping,
+                dataset_mapping[mapping.dimension_type],
+                project_mapping[mapping.dimension_type],
+            )
+            mapping_tables.append(mapping_table)
+
+        mappings_config = DimensionMappingsConfig.load_from_model(
+            DimensionMappingsConfigModel(mappings=mapping_tables), src_dir
+        )
+        mapping_ids = self._dimension_mapping_mgr.register_from_config(
+            mappings_config, submitter, log_message, context=context
+        )
+        for mapping_id in mapping_ids:
+            mapping_config = self._dimension_mapping_mgr.get_by_id(mapping_id)
+            references.append(
+                DimensionMappingReferenceModel(
+                    from_dimension_type=mapping_config.model.from_dimension.dimension_type,
+                    to_dimension_type=mapping_config.model.to_dimension.dimension_type,
+                    mapping_id=mapping_id,
+                    version=str(self._dimension_mapping_mgr.get_current_version(mapping_id)),
+                )
+            )
+
+        return references
+
+    def _auto_register_reverse_supplemental_mappings(
```

Comment:
I agree with @lixiliu that this should be done based on a flag rather than completely auto. Not sure if that suggests a name change here or not.